index.d.ts

```diff
@@ -1,219 +1,278 @@
 export = Analytics;
+/**
+ * Represents a generic object in the APIs
+ * Use for parameters like context, traits etc.
+ */
+interface apiObject {
+  [index: string]:
+    | string
+    | number
+    | boolean
+    | apiObject
+    | (string | number | boolean | apiObject)[];
+}
+
+/**
+ * Represents the integration options object
+ * Example usages:
+ * integrationOptions { All: false, "Google Analytics": true, "Braze": true}
+ * integrationOptions { All: true, "Chartbeat": false, "Customer.io": false}
+ */
+interface integrationOptions {
+  // Defaults to true
+  All?: boolean;
+  // Destination name: true/false
+  [index: string]: boolean | undefined;
```

Comment:
We need to extend the same support as the JS SDK for this

Reply:
In JS SDK that is used by the device mode integrations. In Node SDK where will that be used?

Reply:
Integration specific options can also be used in cloud-mode integrations

Reply:
Once implemented in the Node SDK will also update the declaration. As of now not updating it.

Reply:
The integrations object functionality is not SDK specific. Refer to Cloud-mode integrations like Amplitude, FB Pixel etc for more information.

Reply:
Got it